### PR TITLE
TYP: check_untyped_defs core.groupby.ops

### PR DESCRIPTION
**Pull request by simonjayhawkins**
pandas\core\groupby\ops.py:889: error: Need type annotation for 'sdata'  [var-annotated]

### CODE REVIEW

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -866,7 +866,7 @@ def _is_indexed_like(obj, axes, axis: int) -> bool:
 # Splitting / application
 
 
-class DataSplitter:
```
Comment (jbrockmendel) [inline pandas/core/groupby/ops.py:869]: does this allow us to be more specific in _get_sorted_data or _chop below?

**Review by jbrockmendel — APPROVED** [review]

File: pandas/core/groupby/ops.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/groupby/ops.py:869]: yes, in general if the return type of a method is dependant on self.data and using the FrameOrSeries TypeVar.

we covered this before https://github.com/pandas-dev/pandas/pull/29456#pullrequestreview-314631656

**Review by jbrockmendel — APPROVED** [review]

File: pandas/core/groupby/ops.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/groupby/ops.py:869]: I think so, but the subsequent changes proved controversial last time.

atm. This PR just 'ties' the output of `_get_sorted_data` which is typed with the FrameOrSeries TypeVar with self.data.

hence ```'Revealed type is 'FrameOrSeries`1'``` for sdata

and mypy no longer reports `pandas\core\groupby\ops.py:889: error: Need type annotation for 'sdata' [var-annotated]`

the return type of _get_sorted_data is the same as self.data.take. And Series.take returns a Series and DataFrame.take returns a DataFrame. (i.e. same type as caller)

(As an aside, take is defined in NDFrame and Series and similar for other indexing methods. What's the policy regarding method placement? Could they be moved from NDFrame to DataFrame?)

will circle back round to this (if not merged b4) and add a commit with some more types to take advantage of making DataSplitter generic. If too controversial, can always roll back.

**Comment by jbrockmendel** [comment]
thanks @simonjayhawkins

**Pull request merged**